components/tools/OmeroWeb/omeroweb/webclient/templates/webclient/base/base_container.html

```diff
@@ -195,6 +195,61 @@
         {% if page_size %}
             var PAGE_SIZE = {{ page_size }};
         {% endif %}
+
+
+        // ** "Open With" config used by E.g. ome.tree.js **
+        WEBCLIENT.OPEN_WITH = [];
+        {% if ome.open_with %}
+            // This is a json dump, so we can add it directly without escaping
+            WEBCLIENT.OPEN_WITH = {{ ome.open_with|safe }};
+        {% endif %}
+        // Helper can be used by 'open with' plugins to add isEnabled()
+        // handlers to the OPEN_WITH object.
+        OME.setOpenWithEnabledHandler = function(label, fn) {
+            // look for label in OPEN_WITH
+            WEBCLIENT.OPEN_WITH.forEach(function(ow){
+                if (ow.label === label) {
+                    ow.isEnabled = function() {
+                        // wrap fn with try/catch, return true by default
+                        var args = Array.from(arguments);
+                        try {
+                            enabled = fn(args);
+                        } catch (e) {
+                            console.log("Open with " + label + ": " + e);
+                            enabled = true;
+                        }
+                        return enabled;
+                    }
+                }
+            });
+        };
+        OME.setOpenWithActionHandler = function(label, fn) {
+            // look for label in OPEN_WITH
+            WEBCLIENT.OPEN_WITH.forEach(function(ow){
+                if (ow.label === label) {
+                    ow.action = function() {
+                        // wrap fn with try/catch, return false if error
+                        var args = Array.from(arguments);
+                        try {
+                            enabled = fn(args);
+                        } catch (e) {
+                            console.log("Open with " + label + ": " + e);
+                            enabled = false;
+                        }
+                        return enabled;
+                    }
+                }
+            });
+        };
+        // Include any JavaScript snippets provided by 'open with' plugins
+        {% for ow in ome.open_with_scripts %}
```

Comment:
If we're thinking about plugin based extensions here which are likely installed via `pip install` and then activated via OMERO.web configuration could we dispense with this paradigm? If there are issues with plugin JavaScript it is immensely difficult to debug when using anonymous self-invoking functions. 

Such a plugin would of course have one or more JavaScript files in its static directories like the current paradigm which could be included on the page and be required to register themselves with the "open with" functionality on page load.